tokens/usecase/tokens_usecase.go

```diff
@@ -349,26 +391,69 @@ func (t *tokensUseCase) getPricesForBaseDenom(ctx context.Context, baseDenom str
 }
 
 func (t *tokensUseCase) getChainScalingFactorMut(precision int) (osmomath.Dec, bool) {
-	result, ok := t.precisionScalingFactorMap[precision]
-	return result, ok
+	result, ok := t.precisionScalingFactorMap.Load(precision)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	v, ok := result.(osmomath.Dec)
+	if !ok {
+		return osmomath.Dec{}, false
+	}
+
+	return v, true
+}
+
+// UpdateAssetsAtHeightInterval updates assets at configured height interval.
+// Internally, it calls LoadTokensFromChainRegistry as a goroutine.
+func (t *tokensUseCase) UpdateAssetsAtHeightInterval(height uint64) {
+	if height%uint64(t.updateAssetsHeightInterval) == 0 {
+		go func() {
+			if err := t.LoadTokensFromChainRegistry(); err != nil {
+				t.logger.Error("error loading tokens from chain registry", zap.Error(err))
+			}
+		}()
+	}
+}
+
+// LoadTokensFromChainRegistry loads tokens from the chain registry into TokensUsecase.
+// Tokens are loaded only if the contents of the chain registry file have changed.
+func (t *tokensUseCase) LoadTokensFromChainRegistry() error {
+	tokens, hash, err := GetTokensFromChainRegistry(t.chainRegistryAssetsFileURL)
+	if err != nil {
+		return err
+	}
+
+	if t.lastLoadTokensFromChainRegistryHash != hash {
+		t.LoadTokens(tokens)
+	}
+
+	return nil
 }
 
 // GetTokensFromChainRegistry fetches the tokens from the chain registry.
 // It returns a map of tokens by chain denom.
-func GetTokensFromChainRegistry(chainRegistryAssetsFileURL string) (map[string]domain.Token, error) {
```

Comment:
Should we cover this function with tests?

Reply:
I think the ROI of covering this method by tests would be low assuming that we diligently unit test other methods.

It is hard to set up the test for this since we would have to mock out the HTTP request stuff. 

We also have higher-level functional tests for this component ensuring that this parses as intended. So I would advise to not spend too much time at this time.